cli/medperf/logging/utils.py

```diff
@@ -0,0 +1,189 @@
+    config_dict = {}
+    for item in dir(config):
+        if item.startswith("__"):
+            continue
+        config_dict[item] = config_vars[item]
+    config_dict = filter_var_dict_for_yaml(config_dict)
+    return config_dict
+
+
+def filter_var_dict_for_yaml(unfiltered_dict):
+    valid_types = (str, dict, list, int, float)
+    filtered_dict = {}
+    for key, value in unfiltered_dict.items():
+        if not isinstance(value, valid_types) and value is not None:
+            try:
+                value = str(value)
+            except Exception:
+                value = "<OBJECT>"
+        if isinstance(value, dict):
+            value = filter_var_dict_for_yaml(value)
+        filtered_dict[key] = value
+
+    return filtered_dict
+
+
+def get_storage_contents():
+    storage_paths = config.storage.copy()
+    storage_paths["credentials_folder"] = {
+        "base": os.path.dirname(config.creds_folder),
+        "name": os.path.basename(config.creds_folder),
+    }
+    ignore_paths = {"datasets_folder", "predictions_folder", "results_folder"}
+    contents = {}
+
+    for pathname, path in storage_paths.items():
+        if pathname in ignore_paths:
+            contents[pathname] = "<REDACTED>"
+            continue
+        full_path = os.path.join(path["base"], path["name"])
+        p = subprocess.Popen(
+            ["ls", "-lR", full_path],
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+            text=True,
+        )
+        output, _ = p.communicate()
+        if p.returncode != 0:
+            contents[pathname] = "Could not retrieve tree for storage paths"
+        contents[pathname] = output
+
+    return contents
+
+
+def get_installed_packages():
+    installed_packages = {}
+    for dist in metadata.distributions():
+        installed_packages[dist.name] = dist.version
+    return installed_packages
+
+
+def get_python_environment_information():
+    environment_info = {
+        "Operating System": platform.system(),
+        "Operating System Release": platform.release(),
+        "Operating System Version": platform.version(),
+        "Python Implementation": platform.python_implementation(),
+        "Python Version": platform.python_version(),
+        "Python Compiler": platform.python_compiler(),
+        "Python Build": platform.python_build(),
+        "Machine Architecture": platform.machine(),
+        "Processor Type": platform.processor(),
+        "Python Executable": sys.executable,
+        "Installed Modules": get_installed_packages(),
+    }
+    return environment_info
+
+
+def get_additional_information():
+    sh_script = os.path.join(os.path.dirname(__file__), "get_host_info.sh")
+    p = subprocess.Popen(
+        ["bash", sh_script],
+        stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE,
+        text=True,
+    )
+    output, _ = p.communicate()
+    if p.returncode != 0:
+        return "Could not execute bash script for additional debug information"
```

Comment:
let's have the output as is even if an error occured. This will be more helpful right?